### PR TITLE
Charge interface gas early (part 1)

### PR DESCRIPTION
**Pull request by axic**
This fixes everything, but `call` and `storageStore`.

### CODE REVIEW

File: src/eei.cpp [hunk 1]
```diff
@@ -618,11 +627,12 @@ namespace hera {
 
       ensureCondition(!(m_msg.flags & EVMC_STATIC), StaticModeViolation, "selfDestruct");
 
+      takeInterfaceGas(GasSchedule::selfdestruct);
```
Comment (axic) [inline src/eei.cpp]: This should come before the ensureCondition.

**Comment by axic** [comment]
> Explanation?

The discussion in #428. Basically these things are free if the conditions are not met without this change.

**Comment by axic** [comment]
@hugo-dc @lrettig tests need to be refilled with this.

**Comment by hugo-dc** [comment]
@axic refilled tests in this PR: https://github.com/ewasm/tests/pull/97

**Comment by gballet** [comment]
I assume that the tests are failing because they haven't been updated

**Pull request merged**